### ISSUE TITLE
[dropdown-item] Render as button if neither HREF or TO provided, or add option to render as button?

### ISSUE THREAD

**Issue by tmorehouse** [bootstrap-vue/bootstrap-vue#371]
BS V4.alpha.6 `.dropdown-items` support both using links or buttons (although there are slight styling issues between the two font-wise, which will be fixed in beta release https://github.com/twbs/bootstrap/issues/22560).
http://v4-alpha.getbootstrap.com/components/dropdowns/#menu-items

Maybe the dropdown-item component should create a `<button class="dropdown-item"></button>` when both `href` and `to` are falsy? rather than using a link tag?.

**Comment by tmorehouse** [comment]
Maybe `b-dropdown-item` should use  `b-button` component?

**Comment by MarZab** [comment]
Looking at bootstrap documentation, buttons are the way to go. They look identical to the normal links so I think we should cohere to that https://v4-alpha.getbootstrap.com/components/dropdowns/#menu-items

**Comment by tmorehouse** [comment]
Although, there are some cases where people would like to use dropdown-items as navigation links as well.  So I think the choice of button or link should be made on the presence of `href` and/or `to`

**Comment by MarZab** [comment]
Here is a fiddle with the options:
https://jsfiddle.net/7htvb7gh/3/

Looks like buttons do add a `outline` while `a role=button` do not. On the other hand button text is unselectable while the anchor tag text is.

It might be best going with a link tho, since it would be less code to maintain and it gives a more consitent result.

**Comment by tmorehouse** [comment]
True.  Although dropdown menu text shouldn't be "selectable".  There are a few tweaks to the buttons in V4 Bootstrap coming soon (see https://github.com/twbs/bootstrap/issues/22560)

Right now the link mixin defaults href to `#` (or if the link is disabled).  So maybe this should be altered to allow href to be `null` or `''` (empty string)

**Comment by tmorehouse** [comment]
I can do a mock-up later today to see what can be done.

**Comment by tmorehouse** [comment]
@MarZab, take  look at how b-btn works: https://github.com/bootstrap-vue/bootstrap-vue/blob/master/lib/components/button.vue

Based on the lack presence of either `to` or `href` a button is rendered.  I think this kind of model might work best for dropdown-item.

**Comment by MarZab** [comment]
That makes sense since you did ask for a button to begin with.

I would argue we stick with the link as default for the dropdown for the same reason. It would look and behave exactly as the router and link variants.

**Comment by tmorehouse** [comment]
Maybe add a `use-button` prop on `dropdown items` if a user wants an explicit button.

**Comment by mosinve** [comment]
Aren't b-dropdownitem already can render as link or as button, depending on the `href `and `to`
```
componentType() {
                return (this.href || this.to) ? 'b-link' : 'button';
            },
```

**Comment by tmorehouse** [comment]
I think this might be caused by/related to new link mixin, which defaults `href` to `#`.

**Comment by mosinve** [comment]
I use `.native` modifier as workaround for no-click problem

**Comment by AndrewBarba** [comment]
@mosinve Can you explain the `.native` fix you did? My projects also started breaking recently due to click events not propagating anymore.

@tmorehouse Any updates on supporting a button over a link?

**Comment by mosinve** [comment]
@AndrewBarba , it's the Vue.js event modifier.
detailed info here: https://vuejs.org/v2/guide/migration.html#Listening-for-Native-Events-on-Components-with-v-on-changed

**Comment by AndrewBarba** [comment]
Thank you, very helpful

**Comment by tmorehouse** [comment]
I believe the original issue should be fixed by PR #520

**Comment by tmorehouse** [comment]
This will be fixed in 0.17.0 with the addition of a new component `b-dropdown-button`
